### PR TITLE
refactor(functions): converting limit to use arrow arrays

### LINKED ISSUE #508: Convert limit to use DoArrow

### PR DESCRIPTION
**Pull request by affo**
Fixes #508 

### Done checklist
- [X] docs/SPEC.md updated
- [X] Test cases written

### CODE REVIEW

File: functions/transformations/limit.go [hunk 1]
```diff
@@ -155,7 +155,7 @@ func (t *limitTransformation) Process(id execute.DatasetID, tbl flux.Table) erro
 		}
 		n -= count
 		lcr := sliceColReader{
-			ColReader: cr,
+			ColReader: arrow.ColReader(cr),
```
Comment (affo) [inline functions/transformations/limit.go]: This seems wrong to me 😅 

Is there any other method that allows to reslice an arrow array and append it as column?  
Should I stick with converting it manually to a slice?

Comment (nathanielc) [reply functions/transformations/limit.go]: The arrow packages has ways to create slices from arrow arrays https://godoc.org/github.com/apache/arrow/go/arrow#example-package--Float64Slice

Comment (jsternberg) [reply functions/transformations/limit.go]: I made a similar comment here: https://github.com/influxdata/flux/pull/566#pullrequestreview-186636287

Comment (affo) [reply functions/transformations/limit.go]: I should learn to browse godocs 😅 Thank you!

**Comment by affo** [comment]
Must be merged after #575

**Comment by affo** [comment]
@jsternberg this PR now uses adam's new testing functions.

**Pull request merged**